File: src/main/java/org/demoth/aworlds/server2/UserService.java
package org.demoth.aworlds.server2;

import org.demoth.aworlds.server2.model.User;
import org.springframework.beans.factory.config.ConfigurableBeanFactory;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Component
@Scope(value = ConfigurableBeanFactory.SCOPE_SINGLETON)
public class UserService {
    private final Map<String, User> currentUsers = new HashMap<>();

    public User login(String user, String pass) {
        if ("demoth".equals(user))
            return new User();
        return null;
    }

    public List<String> register(User user, String id) {
        currentUsers.put(id, user);
        ArrayList<String> chars = new ArrayList<>();
        chars.add("Totemy [Templar lvl 12]");
        chars.add("ArkWi4ka [Witch lvl 2]");
        return chars;
    }
}

File: src/main/kotlin/org/demoth/gameserver/UserService.kt
package org.demoth.gameserver

import org.demoth.gameserver.model.User
import org.springframework.beans.factory.config.ConfigurableBeanFactory
import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component
import java.util.*

@Component
@Scope(value = ConfigurableBeanFactory.SCOPE_SINGLETON)
open class UserService {
    private val currentUsers = HashMap<String, User>()

    fun login(user: String, pass: String): User? {
        return if (user == "test" && pass == "test") User() else null
    }

    fun register(user: User, id: String): List<String> {
        currentUsers[id] = user
        val chars = ArrayList<String>()
        chars.add("test character 1")
        chars.add("test character 2")
        return chars
    }
}
